lib/google_crawler/search/search_keyword_worker.ex

```diff
@@ -0,0 +1,84 @@
+defmodule GoogleCrawler.SearchKeywordWorker do
+  use GenServer
+
+  alias GoogleCrawler.Search
+  alias GoogleCrawler.Search.Keyword
+
+  @max_retry_count 3
+
+  # Client
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def search(keyword_id) do
+    GenServer.call(__MODULE__, {:search, keyword_id})
+  end
+
+  # Server Callbacks
+
+  def init(state) do
+    {:ok, state}
+  end
+
+  def handle_call({:search, keyword_id}, _from, state) do
+    IO.puts("Handle call #{keyword_id}")
+
+    keyword = Search.get_keyword(keyword_id)
+    Search.update_keyword(keyword, %{status: :in_progress})
+
+    # start task and store the state of the task with the retry count
+    # as a tuple of {keyword, retry_count} -> {%Keyword{}, 1}
+    task = start_task(keyword)
+    new_state = Map.put(state, task.ref, {keyword, 0})
+
+    {:reply, :ok, new_state}
+  end
+
+  def handle_info({ref, result}, state) do
```

Comment:
How/when does `handle_info` gets called? 🤔 `start_task` is started without any links so how?

Reply:
For this one,

it happens when we start the task with `Task.Supervisor.async_nolink` below. it will send the message with task reference and the result back here.

But if the task is failed, it will trigger the `handle_info({:DOWN, ref, :process, _pid, _reason}, state)` below.

More info:
https://hexdocs.pm/elixir/Task.Supervisor.html#async_nolink/3

Reply:
I understood more by seeing the actual [code example](https://hexdocs.pm/elixir/Task.Supervisor.html#async_nolink/3-examples). 

At first, I was confused as I thought that the supervisor would be notified, not the Genserver. For the supervisor to be notified, then it requires to use `start_child`.